sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -325,14 +325,17 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
         .getOrElse(Array.empty[String])
     val bucketSpec = Option(ctx.bucketSpec()).map(visitBucketSpec)
 
+    // TODO: this may be wrong for non file-based data source like JDBC, which should be external
+    // even there is no `path` in options. We should consider allow the EXTERNAL keyword.
+    val tableType = if (new CaseInsensitiveMap(options).contains("path")) {
+      CatalogTableType.EXTERNAL
```

Comment:
We can remove https://github.com/cloud-fan/spark/blob/0e9b3d2a81f0251f710eae42adca307cd9c69d33/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala#L316-L318 ?

After this PR, it becomes much clear to users. We can support `Create External Data Source Table`! We just need to confirm the `path` is provided.